analyzers/src/SonarAnalyzer.CSharp/Rules/CloudNative/AzureFunctionsLogFailures.cs

```diff
@@ -0,0 +1,189 @@
+                ? methodSymbol
+                : null;
+        }
+
+        private sealed class LoggerCallWalker : SafeCSharpSyntaxWalker
+        {
+            private static readonly int[] ValidLogLevel = new[]
+            {
+                3, // Warning
+                4, // Error
+                5, // Critical
+            };
+
+            private readonly ImmutableHashSet<ExpressionSyntax>.Builder invalidIvocationsBuilder = ImmutableHashSet.CreateBuilder<ExpressionSyntax>();
+
+            public LoggerCallWalker(ILanguageFacade languageFacade, SemanticModel model, ITypeSymbol iLoggerSymbol, CancellationToken cancellationToken)
+            {
+                Language = languageFacade;
+                Model = model;
+                ILogger = iLoggerSymbol;
+                CancellationToken = cancellationToken;
+                LoggerExtensions = new Lazy<INamedTypeSymbol>(() => Model.Compilation.GetTypeByMetadataName(KnownType.Microsoft_Extensions_Logging_LoggerExtensions.TypeName));
+                ILogger_Log = new Lazy<IMethodSymbol>(() => ILogger.GetMembers().OfType<IMethodSymbol>().FirstOrDefault(x => x.Name == "Log"));
+            }
+
+            private ILanguageFacade Language { get; }
+            private SemanticModel Model { get; }
+            private ITypeSymbol ILogger { get; }
+            private CancellationToken CancellationToken { get; }
+            private Lazy<INamedTypeSymbol> LoggerExtensions { get; }
+            private Lazy<IMethodSymbol> ILogger_Log { get; }
+            public bool HasValidLoggerCall { get; private set; }
+            public ImmutableArray<ExpressionSyntax> InvalidLoggerInvocations => invalidIvocationsBuilder.ToImmutableArray();
+
+            public override void Visit(SyntaxNode node)
+            {
+                CancellationToken.ThrowIfCancellationRequested();
+                if (HasValidLoggerCall)
+                {
+                    return;
+                }
+                base.Visit(node);
+            }
+
+            public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+            {
+                if (Model.GetSymbolInfo(node, CancellationToken).Symbol is IMethodSymbol { ReceiverType: { } receiver }
+                    && receiver.DerivesOrImplements(ILogger))
+                {
+                    if (IsValidLogCall(node))
+                    {
+                        HasValidLoggerCall = true;
+                    }
+                    else
+                    {
+                        invalidIvocationsBuilder.Add(node);
+                    }
+                }
+                base.VisitInvocationExpression(node);
+            }
+
+            private bool IsValidLogCall(InvocationExpressionSyntax invocation)
+            {
+                if (Model.GetSymbolInfo(invocation, CancellationToken).Symbol is IMethodSymbol symbol)
```

Comment:
This is querying `IMethodSymbol` for second time, it was already evaluated in `VisitInvocationExpression`